### PR TITLE
taproot address validation

### PR DESCRIPTION
**Pull request by xsats**
Add validation logic for taproot/unknown segwit versions. Thanks to bitcoinjs/BlueWallet

### CODE REVIEW

**Comment by xsats** [comment]
AFAIK this should also add support for sending to bech32m addresses through the upgrate to `bitcoinjs-lib` v^6.0.0.

**Comment by BitcoinZavior** [comment]
@xsats can you provide the .env file you used to run the test cases. I can test your branch add add the results here.

**Comment by xsats** [comment]
Hey @BitcoinZavior! Unfortunately I don't have access to the original .env file either so I'm unable to share the missing mnemonics. 

@tanx - are you able to share this? Test scripts work for the specific changes introduced by my two PRs, which use the hardcoded mnemonics but obviously would be great to validate the entire suite without rewriting the results.

**Comment by tanx** [comment]
> @tanx - are you able to share this? Test scripts work for the specific changes introduced by my two PRs, which use the hardcoded mnemonics but obviously would be great to validate the entire suite without rewriting the results.

See https://github.com/photon-sdk/photon-lib/issues/33#issuecomment-1025169480

**Comment by xsats** [comment]
@tanx what are your plans for Photon going forwards? Do you need help with review of PRs/maintaining the repo?

**Comment by tanx** [comment]
> @tanx what are your plans for Photon going forwards? Do you need help with review of PRs/maintaining the repo?

Yes, I'm happy to hand over maintenance of the project if someone steps up to take ownership of reviewing PRs, etc.

**Comment by tanx** [comment]
Also, I'm happy to do a handover/transition and offer some mentorship during this period.

**Comment by xsats** [comment]
Great - more than happy to help maintain the project and very grateful for any of your guidance during the transition. @BitcoinZavior also expressed interest on this so happy to work with you both

File: src/wallet/multisig-hd-wallet.js [hunk 1]
```diff
@@ -886,7 +886,8 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
               seed = MultisigHDWallet.convertElectrumMnemonicToSeed(cosigner);
             }
 
-            const hdRoot = HDNode.fromSeed(seed);
+
```
Comment (tanx) [inline src/wallet/multisig-hd-wallet.js:889]:
```suggestion
```

**Review by tanx — CHANGES_REQUESTED** [review]
LGTM ... just a minor nit

**Comment by tanx** [comment]
> LGTM ... just a minor nit

Ah wait sorry, just looked at the last commit. Will take another look for the rest.

**Comment by tanx** [comment]
I gave the whole PR another review. Generally looks good. The tests are failing... likely since it's based on an outdated master. Can you rebase to the current master and then we can see if tests pass? Thanks 🙏

**Pull request closed without merging**